crates/router/src/connector/braintree/transformers.rs

```diff
@@ -228,17 +228,17 @@ impl<F, T>
             types::PaymentsResponseData,
         >,
     ) -> Result<Self, Self::Error> {
+        let connector_id =
+            types::ResponseId::ConnectorTransactionId(item.response.transaction.id.clone());
```

Comment:
```suggestion
        let id =
            item.response.transaction.id.clone();
```
Please use this id  as both resource_id and connector_response_reference_id